### PR TITLE
TW-1356: Reword accounts picker

### PR DESCRIPTION
**Pull request by sherlockvn**
### Root cause:
The currentProfileNotifier is updated multiple times during the build of chat list, and when its first build, the `_activeClient` default is first client in the all clients.

-> Solution: move `currentProfileNotifier` to only the TwakeHeader, and make sure it avatar updated only one if have the new change in avatar or the inital build. Add the condition for check the `_activeClient` for update the avatar correctly

### Resolved (23 - May)

https://github.com/linagora/twake-on-matrix/assets/99852347/db435564-50bf-4998-b886-3e3109f6b1d6

- Switch and logout

https://github.com/linagora/twake-on-matrix/assets/99852347/d5b83890-204e-42d4-a0da-42661bbbfe25

### CODE REVIEW

File: lib/pages/multiple_accounts/multiple_accounts_picker.dart [hunk 1]
```diff
@@ -82,10 +80,9 @@ class MultipleAccountsPickerController {
       titleAccountSettings: L10n.of(context)!.accountSettings,
       logoApp: Padding(
         padding: TwakeHeaderStyle.logoAppOfMultiplePadding,
-        child: SvgPicture.asset(
-          ImagePaths.icTwakeImageLogo,
-          width: TwakeHeaderStyle.logoAppOfMultipleWidth,
-          height: TwakeHeaderStyle.logoAppOfMultipleHeight,
```
Comment (imGok) [inline lib/pages/multiple_accounts/multiple_accounts_picker.dart:88]: you should delete these from style file

Comment (sherlockvn) [reply lib/pages/multiple_accounts/multiple_accounts_picker.dart:88]: done

**Review by Te-Z — APPROVED** [review]

**Review by nqhhdev — DISMISSED** [review]

**Review by hoangdat — DISMISSED** [review]
@sherlockvn please try to test it with current branch

**Comment by nqhhdev** [comment]
Pls upload new demo (IOS, Mobile) for this case.

**Comment by hoangdat** [comment]
- please check this case 
image

**Review by nqhhdev — APPROVED** [review]

**Comment by hoangdat** [comment]
image

File: lib/pages/settings_dashboard/settings_profile/settings_profile_view.dart [hunk 2]
```diff
@@ -75,7 +75,7 @@ class SettingsProfileView extends StatelessWidget {
       backgroundColor: responsive.isWebDesktop(context)
           ? Theme.of(context).colorScheme.surface
           : LinagoraSysColors.material().onPrimary,
-      body: SingleChildScrollView(
+      body: Padding(
```
Comment (nqhhdev) [inline lib/pages/settings_dashboard/settings_profile/settings_profile_view.dart:78]: Why need to change padding ?

File: lib/pages/settings_dashboard/settings_profile/settings_profile_view_mobile_style.dart [hunk 3]
```diff
@@ -16,4 +16,13 @@ class SettingsProfileViewMobileStyle {
...
       const EdgeInsetsDirectional.all(8);
+
+  static const bottomButtonHeight = 48.0;
+  static const paddingBottomButton = EdgeInsets.only(left: 16.0, right: 16.0);
+  static const bottomButtonRadius = 100.0;
+  static const iconSize = 18.0;
+
+  static const paddingIconAndText = SizedBox(width: 8.0);
+
+  static const paddingBottomBottomButton = SizedBox(height: 32.0);
```
Comment (nqhhdev) [inline lib/pages/settings_dashboard/settings_profile/settings_profile_view_mobile_style.dart:27]: WTF?

File: lib/pages/settings_dashboard/settings_profile/settings_profile_view_mobile.dart [hunk 4]
```diff
@@ -35,167 +40,218 @@ class SettingsProfileViewMobile extends StatelessWidget {
...
-            ],
+                    ),
+                  ],
+                );
+              },
+            ),
           ),
         ),
-        settingsProfileOptions,
+        SettingsProfileViewMobileStyle.paddingBottomBottomButton,
```
Comment (nqhhdev) [inline lib/pages/settings_dashboard/settings_profile/settings_profile_view_mobile.dart:254]: Why you don't add bottom margin in `Container`

**Comment by nqhhdev** [comment]
https://github.com/linagora/twake-on-matrix/assets/99852347/6a5f7f74-f9c8-4b16-be24-c559bcdddef5

I found some issue:

- When I click on add another account , some time lagging 
- Open settings → click on profile → add another account → login success → back to settings screen ??? IMO: why don't back to chat list screen?

**Comment by nqhhdev** [comment]
Issue related to Back button in account picker


https://github.com/linagora/twake-on-matrix/assets/99852347/4bafdf76-0c02-4839-921f-a936caf9af41

1. Click on icon avatar in chat list → open profile → click on add another account → open homeserver picker → I click on back button → back to chat list.
  - Why don't back to profile?

2. Open settings → Open profile → Click on add another account → open homeserver picker → I click on back button → back to settings.
   - Why don't back to settings tab?
   - Still highlight profile button?

File: lib/pages/settings_dashboard/settings_profile/settings_profile.dart [hunk 5]
```diff
@@ -583,6 +603,9 @@ class SettingsProfileController extends State<SettingsProfile>
   void initState() {
     _handleViewState();
     _getCurrentProfile(client);
+    WidgetsBinding.instance.addPostFrameCallback((timeStamp) async {
+      haveMultipleAccountsNotifier.value = await accountsCount > 1;
+    });
```
Comment (nqhhdev) [inline lib/pages/settings_dashboard/settings_profile/settings_profile.dart:608]: No good for me, UI/UX not smoothly. IMO: You should define state for this. If during get all clients show loading in button


https://github.com/linagora/twake-on-matrix/assets/99852347/f5267e26-15e0-42d9-a63c-798bb3bd7851

**Comment by nqhhdev** [comment]
Issue released to switch account:

https://github.com/linagora/twake-on-matrix/assets/99852347/235e8492-679a-4c92-8a90-79539da4e701


-  Open settings → Open profile → Switch to another account → switch account success back to settings.
   - Why need to back settings? Why don't back to profile?

**Comment by hoangdat** [comment]
- [ ] we still not support multiple account in web app

**Comment by sherlockvn** [comment]
i tested the case log out again and it works well

**Comment by Te-Z** [comment]
> * [ ]  we still not support multiple account in web app
> 
> image

IMO we should create a ticket to support this feature on web. WDYT ?

**Pull request merged**